src/main/java/svenhjol/charm/crafting/block/BlockLantern.java

```diff
@@ -189,17 +200,46 @@ protected BlockStateContainer createBlockState()
     @Override
     public IBlockState getStateForPlacement(World world, BlockPos pos, EnumFacing facing, float hitX, float hitY, float hitZ, int meta, EntityLivingBase placer)
     {
-        if (facing.equals(EnumFacing.DOWN)) {
+        if (canPlaceBelow(world, pos)) {
             return this.getDefaultState().withProperty(HANGING, true);
         }
-
         return this.getDefaultState().withProperty(HANGING, false);
     }
 
     @Override
-    public boolean canPlaceBlockOnSide(World world, BlockPos pos, EnumFacing side)
+    public boolean canPlaceBlockAt(World world, @Nonnull BlockPos pos)
     {
-        return side.equals(EnumFacing.DOWN) || side.equals(EnumFacing.UP);
+        return super.canPlaceBlockAt(world, pos) && (canPlaceAbove(world, pos) || canPlaceBelow(world, pos));
+    }
+
+    private boolean canPlaceAbove(World world, BlockPos pos)
+    {
+        IBlockState stateBelow = world.getBlockState(pos.down());
+        Block blockBelow = stateBelow.getBlock();
+
+        return blockBelow == Blocks.LIT_PUMPKIN || (blockBelow != Blocks.AIR && blockBelow.canPlaceTorchOnTop(stateBelow, world, pos.down()));
```

Comment:
What's the deal with `LIT_PUMPKIN`? :)

Reply:
The method `canPlaceTorchOnTop` returns false if the block below is `LIT_PUMPKIN`, so I thought it'd make sense to include it as well as it's a solid block.